### PR TITLE
Test Localization Rule #16153

### LINKED ISSUE #16153: /Localization/[CultureName]/[ModuleId].po PO file location doesn't work
### To Reproduce
 Hi all, I've found that this rule mentioned in this document https://docs.orchardcore.net/en/latest/reference/modules/Localize/ doesn't seem to work
`/Localization/[CultureName]/[ModuleId].po`
This is a module based on OC 1.8.3
image

But when I use `Localisation/zh-CN.po` it works fine

Since this template `Localisation/[CultureName].po`  will overwrite the translation files of the other modules when compiled and packaged.
I would prefer that the template `/Localisation/[CultureName]/[ModuleId].po` works.
When I run it on 1.8x it doesn't work as expected.

### PR DESCRIPTION
**Pull request by hyzx86**
For reproduction  issue #16153

### CODE REVIEW

**Comment by hyzx86** [comment]
Hi @Piedone , Is it possible to reference OrchardCore.Demo project directly in our unit test project?
This PR is to validate the internationalisation file rules in the custom module

**Comment by hyzx86** [comment]
> The Demo project is not really for use cases like that.

 Okay, looks like I should add the test code to `ModuleSample`.
image

File: test/OrchardCore.Tests/Localization/DefaultPluralRuleProviderTests.cs [hunk 1]
```diff
@@ -25,5 +28,28 @@ public void TryGetRuleShouldReturnRuleForTopCulture(string culture, string expec
...
+                 await shellFeaturesManager.EnableFeaturesAsync([demoFeature], true);
+
+                 var shellSettings = scope.ServiceProvider.GetRequiredService<ShellSettings>();
+                 var shellHost = scope.ServiceProvider.GetRequiredService<IShellHost>();
+                 await shellHost.ReloadShellContextAsync(shellSettings);
+             });
+
+            var response = await context.Client.GetAsync("api/demo/sayhello");
+
+            response.EnsureSuccessStatusCode();
```
Comment (hyzx86) [inline test/OrchardCore.Tests/Localization/DefaultPluralRuleProviderTests.cs:52]: Hi @hishamco I'm trying to send a request here but it keeps returning 404 , I can't find out why, can you help me see what's going on?Thanks!

This PR is not meant to be merged into the main branch, it's just meant to replicate the translation issue

Comment (hishamco) [reply test/OrchardCore.Tests/Localization/DefaultPluralRuleProviderTests.cs:52]: Do you mean to reproduce the issue?

Comment (hyzx86) [reply test/OrchardCore.Tests/Localization/DefaultPluralRuleProviderTests.cs:52]: > Do you mean to reproduce the issue?

yes

File: src/OrchardCore.Modules/OrchardCore.Localization/ModularPoFileLocationProvider.cs [hunk 2]
```diff
@@ -74,6 +74,9 @@ public IEnumerable<IFileInfo> GetLocations(string cultureName)
 
                 // \src\OrchardCore.Cms.Web\Localization\OrchardCore.Cms.Web-fr-CA.po
                 yield return _fileProvider.GetFileInfo(PathExtensions.Combine(_resourcesContainer, extension.Id + CultureDelimiter + poFileName));
+
+                // \Localization\[CultureName]\[ModuleId].po
+                yield return _fileProvider.GetFileInfo(PathExtensions.Combine(_resourcesContainer, cultureName, extension.Id + PoFileExtension));
```
Comment (hyzx86) [inline src/OrchardCore.Modules/OrchardCore.Localization/ModularPoFileLocationProvider.cs:79]: We do not support the format `/Localisation/[CultureName]/[ModuleId].po` as mentioned in the [documentation](https://docs.orchardcore.net/en/latest/reference/modules/Localize/).

**Pull request closed without merging**